app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,28 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @post = Post.new(
+      title: params[:post][:title],
+      text: params[:post][:text]
+    )
+    @post.user_id = params[:user_id]
```

Comment:
- [ ] Please use the `current_user` method here to identify the post's owner.
`@post = current_user.posts.new(title: params[:title],
      text: params[:text])`
OR
`@post = current_user.posts.new(post_params)` if you used the `params` method as requested as optional.